### PR TITLE
[mlir][func] Remove `func-bufferize` pass

### PR DESCRIPTION
**Pull request by matthias-springer**
The dialect conversion-based bufferization passes have been migrated to One-Shot Bufferize about two years ago. To clean up the code base, this commit removes the `func-bufferize` pass, one of the few remaining parts of the old infrastructure. Most bufferization passes have already been removed.

Note for LLVM integration: If you depend on this pass, migrate to One-Shot Bufferize or copy the pass to your codebase.

Depends on #114017.

### CODE REVIEW

File: mlir/test/Dialect/Transform/test-pass-application.mlir [hunk 1]
```diff
@@ -111,9 +111,9 @@ module attributes {transform.with_named_sequence} {
   transform.named_sequence @__transform_main(%arg1: !transform.any_op) {
     %1 = transform.structured.match ops{["func.func"]} in %arg1 : (!transform.any_op) -> !transform.any_op
 
-    // func-bufferize can be applied only to ModuleOps.
+    // duplicate-function-elimination can be applied only to ModuleOps.
     // expected-error @below {{pass pipeline failed}}
-    transform.apply_registered_pass "func-bufferize" to %1 : (!transform.any_op) -> !transform.any_op
+    transform.apply_registered_pass "duplicate-function-elimination" to %1 : (!transform.any_op) -> !transform.any_op
```
Comment (javedabsar1) [inline mlir/test/Dialect/Transform/test-pass-application.mlir:116]: AFAIU here 'func-bufferize' just happened to be used. It is not really playing any specific role. right?

Comment (matthias-springer) [reply mlir/test/Dialect/Transform/test-pass-application.mlir:116]: That's right. This test tries to apply a ModulePass to a function and checks if an error is produced. Any ModulePass will do for the test.

**Comment by matthias-springer** [comment]
@javedabsar1 Is this good to merge?

**Review by javedabsar1 — APPROVED** [review]
LGTM. The implementation (partial bufferization) is being deleted so no point keeping documentation for it.

**Pull request merged**